### ISSUE TITLE
Fed canary: when multiple clusters are registered with same name, fed writes to wrong cluster

### ISSUE THREAD

**Issue by jenshu** [solo-io/gloo#7529]
### Gloo Edge Version

1.12.x (latest stable)

### Kubernetes Version

None

### Describe the bug

When 2 Gloo Fed instances are installed on the same management cluster, and each registers a Gloo Edge instance on a remote cluster under the same cluster name, Gloo Fed sometimes tries to federate resources to the wrong cluster.

### Steps to reproduce the bug

Create 3 clusters (mgmt, remote1, remote2)
```
kind create cluster --name mgmt
kind create cluster --name remote1
kind create cluster --name remote2
```
Install 2 instances of Gloo Fed on mgmt in namespaces gloo-system-old and gloo-system-new
```
k config use-context kind-mgmt
helm install -n gloo-system-old gloo-fed gloo-fed/gloo-fed --create-namespace --set-string license_key=$GLOO_LICENSE_KEY
helm install -n gloo-system-new gloo-fed gloo-fed/gloo-fed --create-namespace --set-string license_key=$GLOO_LICENSE_KEY
```
Install GlooEE on remote1 in gloo-system namespace
```
k config use-context kind-remote1
helm install -n gloo-system gloo-ee gloo-ee/gloo-ee --create-namespace --set-string license_key=$GLOO_LICENSE_KEY --set gloo-fed.enabled=false
```
Install GlooEE on remote2 in gloo-system namespace
```
k config use-context kind-remote2
helm install -n gloo-system gloo-ee gloo-ee/gloo-ee --create-namespace --set-string license_key=$GLOO_LICENSE_KEY --set gloo-fed.enabled=false
```
On mgmt cluster, register both GlooEE instances (note that the clusters are registered with the same name and remote namespace)
```
k config use-context kind-mgmt
glooctl cluster register --federation-namespace gloo-system-old --remote-namespace gloo-system --cluster-name mycluster --remote-context kind-remote1 --local-cluster-domain-override host.docker.internal
glooctl cluster register --federation-namespace gloo-system-new --remote-namespace gloo-system --cluster-name mycluster --remote-context kind-remote2 --local-cluster-domain-override host.docker.internal
```
Create a FederatedUpstream
```
k create ns gloo-fed
kubectl apply -f - <<EOF
apiVersion: fed.gloo.solo.io/v1
kind: FederatedUpstream
metadata:
  name: my-fed-upstream
  namespace: gloo-fed
spec:
  placement:
    clusters:
      - mycluster
    namespaces:
      - gloo-system
  template:
    metadata:
      name: my-upstream
    spec:
      kube:
        serviceName: my-service
        serviceNamespace: default
        servicePort: 10000
EOF
```
Check the placement status of the FederatedUpstream (it shows placed for both gloo-system-old and gloo-system-new)
```
k get federatedupstream -n gloo-fed my-fed-upstream -oyaml
```
Check that the upstream has been created on both remote clusters
```
k --context kind-remote1 get us -n gloo-system my-upstream -oyaml
k --context kind-remote2 get us -n gloo-system my-upstream -oyaml
```
Note that the upstream has only been created on one cluster.

### Expected Behavior

The upstream should have been successfully created on both remote clusters.
One of the Fed controllers incorrectly updated the upstream on the wrong cluster.

**Comment by jenshu** [comment]
depends on solo-io/skv2#399

**Comment by jenshu** [comment]
fix available in GlooEE v1.14.0-beta2, v1.13.0-rc4, and v1.12.39
